src/core/query.ts

```diff
@@ -363,7 +379,7 @@ export class Query<
 
     // Create query function context
     const queryKey = ensureArray(this.queryKey)
-    const queryFnContext: QueryFunctionContext = {
+    const queryFnContext: QueryFunctionContext<unknown[]> = {
```

Comment:
I needed to use `unknown[]` here because after `ensureArray`, `TQueryKey` is no longer true, as that contains strings as well. I think it's not a big deal, but if you know how to make `ensureArray` return `TQueryKey` if the input is an array and `[string]` is not, we could improve this further.